### ISSUE TITLE
relay-compiler has implicit dependency on lodash>=4 and core-js<3

### ISSUE THREAD

**Issue by wyattanderson** [facebook/relay#2759]
Hi there! I've been working on upgrading our codebase to Relay 4.0.0 and ran into a few roadblocks. It seems that `relay-compiler` has an implicit (i.e. not specified in `package.json` and therefore not resolved by `npm install` when installing `relay-compiler`) dependency on `lodash>=4`, and that if your application explicitly pins `lodash<4` (for historical reasons, let's not dive into that), `relay-compiler` can't run at all because it can't import `lodash/clone`:

```
ERROR:
Error writing modules:
Error: Cannot find module 'lodash/clone'
    at Function.Module._resolveFilename (module.js:538:15)
    at Function.Module._load (module.js:468:25)
    at Module.require (module.js:587:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/wyatt/git/relay-lodash-reproducer/node_modules/relay-compiler/bin/relay-compiler:24273:18)
    at __webpack_require__ (/Users/wyatt/git/relay-lodash-reproducer/node_modules/relay-compiler/bin/relay-compiler:30:30)
    at _clone (/Users/wyatt/git/relay-lodash-reproducer/node_modules/relay-compiler/bin/relay-compiler:24226:39)
    at keys.forEach.key (/Users/wyatt/git/relay-lodash-reproducer/node_modules/relay-compiler/bin/relay-compiler:24257:38)
    at Array.forEach (<anonymous>)
    at builder (/Users/wyatt/git/relay-lodash-reproducer/node_modules/relay-compiler/bin/relay-compiler:24253:8)
```

I've built a minimal reproducer at https://github.com/wyattanderson/relay-lodash-reproducer which you can trigger by cloning, running `npm install`, and then running `node_modules/.bin/relay-compiler  --extensions jsx --schema schema.graphql  --src src/`, which will crash with the above error.

And, there's a bonus! On the `core-js` branch in that repository, I've pinned `core-js@3.1.3` (the latest version), and the same steps reproduce the issue in #2701:

```
Error: Cannot find module 'core-js/es6'
    at Function.Module._resolveFilename (module.js:538:15)
    at Function.Module._load (module.js:468:25)
    at Module.require (module.js:587:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/wyatt/git/relay-lodash-reproducer/node_modules/relay-compiler/bin/relay-compiler:15037:18)
    at __webpack_require__ (/Users/wyatt/git/relay-lodash-reproducer/node_modules/relay-compiler/bin/relay-compiler:30:30)
    at Object.<anonymous> (/Users/wyatt/git/relay-lodash-reproducer/node_modules/relay-compiler/bin/relay-compiler:9663:1)
    at __webpack_require__ (/Users/wyatt/git/relay-lodash-reproducer/node_modules/relay-compiler/bin/relay-compiler:30:30)
    at Object.<anonymous> (/Users/wyatt/git/relay-lodash-reproducer/node_modules/relay-compiler/bin/relay-compiler:14950:1)
    at __webpack_require__ (/Users/wyatt/git/relay-lodash-reproducer/node_modules/relay-compiler/bin/relay-compiler:30:30)
```

**Comment by wyattanderson** [comment]
This is still an issue with 5.0.0.

**Comment by sibelius** [comment]
check here https://github.com/facebook/relay/search?q=lodash%2Fclone&unscoped_q=lodash%2Fclone

it is only a dev dependency on lodash when running 

yarn why

```jsx
sibelius-pro:relay sibelius$ yarn why lodash/clone
yarn why v1.16.0
[1/4] 🤔  Why do we have the module "lodash/clone"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "lodash@4.17.11"
info Reasons this module exists
   - "eslint-plugin-flowtype" depends on it
   - Hoisted from "eslint-plugin-flowtype#lodash"
   - Hoisted from "@babel#core#lodash"
   - Hoisted from "@babel#types#lodash"
   - Hoisted from "@babel#traverse#lodash"
   - Hoisted from "@babel#generator#lodash"
   - Hoisted from "eslint#lodash"
   - Hoisted from "babel-preset-fbjs#@babel#plugin-transform-block-scoping#lodash"
   - Hoisted from "@babel#plugin-transform-modules-commonjs#@babel#helper-module-transforms#lodash"
   - Hoisted from "eslint#table#lodash"
   - Hoisted from "eslint#inquirer#lodash"
   - Hoisted from "babel-preset-fbjs#@babel#plugin-transform-classes#@babel#helper-define-map#lodash"
   - Hoisted from "jest#jest-cli#jest-config#jest-environment-jsdom#jsdom#request-promise-native#request-promise-core#lodash"
info Disk size without dependencies: "4.86MB"
info Disk size with unique dependencies: "4.86MB"
info Disk size with transitive dependencies: "4.86MB"
info Number of shared dependencies: 0
✨  Done in 1.04s.
```

not sure why is this throwing for you

**Comment by robrichard** [comment]
might be a symptom of the same issue i described here: https://github.com/facebook/relay/issues/2698#issuecomment-476361205

**Comment by wyattanderson** [comment]
> might be a symptom of the same issue i described here: [#2698 (comment)](https://github.com/facebook/relay/issues/2698#issuecomment-476361205)

I think that might be it, at least for me! Thanks! Hoping that #2764 fixes it.

**Comment by kassens** [comment]
Something is broken in the webpack config we use.

After the build, the dist/relay-compiler file contains a bunch of `require('lodash/.*)` that I don't know why they're there. I think those should all be inlined during compilation?

**Comment by kassens** [comment]
Just saw #2764, that makes sense to me. Fingers crossed that fixes it!

**Comment by kassens** [comment]
This should be fixed in the next release thanks to #2764.

**Comment by wyattanderson** [comment]
All credit to @robrichard for tracking down the root cause!

@kassens any chance of a minor release in the near future? This is blocking us from upgrading past 1.7.0.
